### PR TITLE
Enable DLDT tests in CI

### PR DESCRIPTION
**Pull request by rblaczkowski**
Enable DLDT backend in CI:

    * small refactor of CI
    * add DLDT building in CI
    * add DLDT:IE backend in test infrastructure
    * firstly add possibility to run but disable DLDT in CI (due to repository transition)

### CODE REVIEW

File: .ci/jenkins/ci.groovy [hunk 1]
```diff
@@ -95,18 +108,34 @@ CONFIGURATION_WORKFLOW = { configuration ->
...
+            --no-tags \
+            --dissociate \
+            --depth 1 \
+            --verbose \
+            ${address} \
+            ${WORKDIR}/${repositoryName}
+    """
+}
+
+def gitSubmoduleUpdate(String repository_name, String exception="") {
```
Comment (mchrusci) [inline .ci/jenkins/ci.groovy:129]: What is parameter "String exception" for?

File: .ci/jenkins/ci.groovy [hunk 2]
```diff
@@ -152,23 +181,33 @@ def runDockerContainer(String imageName) {
...
     String backendEnvVar = "NGRAPH_BACKEND=${backend.toUpperCase()}"
-    sh """
-        NGRAPH_WHL=\$(docker exec ${DOCKER_CONTAINER_NAME} find ${DOCKER_HOME}/ngraph/python/dist/ -name 'ngraph*.whl')
-        docker exec -e ${toxEnvVar} -e ${backendEnvVar} -w ${DOCKER_HOME}/ngraph-onnx ${DOCKER_CONTAINER_NAME} \
-            tox -c .
-    """
+    String libraryVar = (backend == "ie") ? "LD_LIBRARY_PATH=${DOCKER_HOME}/dldt_dist/deployment_tools/inference_engine/external/tbb/lib:${DOCKER_HOME}/dldt_dist/deployment_tools/inference_engine/lib/intel64:${DOCKER_HOME}/dldt_dist/deployment_tools/inference_engine/external/mkltiny_lnx/lib:${DOCKER_HOME}/dldt_dist/deployment_tools/ngraph/lib" : "LD_LIBRARY_PATH="
+   
+
+    if (backend == "ie") {
```
Comment (mchrusci) [inline .ci/jenkins/ci.groovy:200]: This doesn't execute unless there's "ie" in `backends` list here (same file, line 23):
```
BACKEND_SKU_CONFIGURATIONS = [
        [ sku : "skx", backends : ["cpu", "interpreter"] ],
        [ sku : "clx", backends : ["cpu", "interpreter"] ],
        [ sku : "bdw", backends : ["cpu", "interpreter"] ]
```

Comment (mchrusci) [reply .ci/jenkins/ci.groovy:200]: Nevermind, I understand this PR just adds possibility to run tests on IE backend that will be switched on later.

File: .ci/jenkins/ci.groovy (hunk 1 above)
Comment (rblaczkowski) [reply .ci/jenkins/ci.groovy:129]: You are right, it is a hangover of excluding to update some submodule of a particular repository. But for now it is useless (not used in definition code), I will fix it.

**Comment by arogowie-intel** [comment]
Unfortunately I'm seeing plenty of segfaults..

**Pull request merged**